File: projetoJava/src/Escola.java
import java.util.ArrayList;
import java.util.List;

class Escola {
    private String nome;
    private List<Professor> professores;

    public Escola(String nome) {
        this.nome = nome;
        this.professores = new ArrayList<>();
    }

    public void adicionarProfessor(Professor professor) {
        if (!professores.contains(professor)) {
            professores.add(professor);
            professor.adicionarEscola(this);
        }
    }

    public void exibirProfessores() {
        System.out.println("Escola " + nome + " tem os seguintes professores:");
        for (Professor professor : professores) {
            System.out.println("- " + professor.getNome());
        }
    }

    public String getNome() {
        return nome;
    }
}

File: projetoGo/escola.go
package main

import "fmt"

type Escola struct {
	nome        string
	professores []*Professor
}

func (e *Escola) AdicionarProfessor(professor *Professor) {
	if !e.temProfessor(professor) {
		e.professores = append(e.professores, professor)
		professor.AdicionarEscola(e)
	}
}

func (e *Escola) temProfessor(professor *Professor) bool {
	for _, p := range e.professores {
		if p == professor {
			return true
		}
	}
	return false
}

func (e *Escola) ExibirProfessores() {
	fmt.Printf("Escola %s tem os seguintes professores:\n", e.nome)
	for _, professor := range e.professores {
		fmt.Printf("- %s\n", professor.nome)
	}
}
